elastic/integrations
# entityanalytics_ad: new package for Active Directory user collection

## Proposed commit message



See title.

## Checklist


## Author's Checklist



## How to test this PR locally



System testing requires an external Active Directory server to collect the user details from. Set this up or get access to an existing server.

This testing depends on 8.14.0-SHAPSHOT including f871b5c5f76b45c5724599d49bd0b0909959783d. The current DRA does not. It is possible to get around this by bringing up the stack as normal with `elastic-package` and inserting a copy of the relevant filebeat into the docker agent filesystem. First start an 8.13.0 stack.

In the x-pack/filebeat directory:
```
PACKAGES=tar.gz PLATFORMS=linux/amd64 DEV=true SNAPSHOT=true mage -v package
tar zxvf build/distributions/filebeat-8.14.0-SNAPSHOT-linux-x86_64.tar.gz filebeat-8.14.0-SNAPSHOT-linux-x86_64/filebeat
elastic-package stack up -v -d --version=8.13.0
docker cp filebeat-8.14.0-SNAPSHOT-linux-x86_64/filebeat <container ID>:/usr/share/elastic-agent/data/elastic-agent-1eb18c/components
```

Then in the integrations repo in the entityanalytics_ad package directory:
``` 
AD_ADDR=<ldap host url> # e.g. ldap://localhost
AD_USER=<ldap user> # e.g. CN=Administrator,CN=Users,DC=testserver,DC=local
AD_PASS=<password>
AD_BASE_DN=<base dn> # e.g. CN=Users,DC=testserver,DC=local
sed -i -r "\
    s|ldapurl|${AD_ADDR}|; \
    s/ldapbasedn/${AD_BASE_DN}/; \
    s/ldapuser/${AD_USER}/; \
    s/ldappassword/${AD_PASS}/; \
" data_stream/user/_dev/test/system/test-default-config.yml
elastic-package stack up -v -d --version=8.14.0-SNAPSHOT # See note above about stack version; do not run this if the filebeat build was required.
eval "$(elastic-package stack shellinit)"
elastic-package test system
```
(assumes GNU `sed`; if on another OS, install and use `gsed`)

## Related issues


- Closes #8559

## Screenshots



![Front page](https://github.com/elastic/integrations/assets/90160302/6b402ed7-30b1-4a70-9206-dc39d4b62095)

![Config UI](https://github.com/elastic/integrations/assets/90160302/693e5959-e8f2-4172-8807-b7e19ffe1a3f)
